test/functional/test_bmc_peer_communication.py

```diff
@@ -0,0 +1,240 @@
+        topo.create()
+        fake_node0_list = self.node_cfg_up('test0', 'node0ns', fixtures.a_boot_image, "dell_r730")
+        fake_node0 = fake_node0_list[0]
+        emu_file_for_del0 = fake_node0_list[1]
+        fake_node1_list = self.node_cfg_up('test1', 'node1ns', fixtures.b_boot_image, "quanta_d51")
+        fake_node1 = fake_node1_list[0]
+        emu_file_for_del1 = fake_node1_list[1]
+
+        """
+        Test peer_bmc fru print
+        """
+        node0_ssh = self.client_ssh('192.168.188.91')
+        status, result0 = node0_ssh.exec_command(cmd_prefix0[0])
+        node0_ssh = self.client_ssh('192.168.188.91')
+        status, result1 = node0_ssh.exec_command(cmd_prefix1[0])
+        self.assertIn("PowerEdge R730", result0, "peer_bmc can not Obtain fru")
+        self.assertIn("Quanta Computer", result1, "peer_bmc can not Obtain fru")
+
+        """
+        Test peer_bmc lan print
+        """
+        node0_ssh = self.client_ssh('192.168.188.91')
+        status, result0 = node0_ssh.exec_command(cmd_prefix0[1])
+        node0_ssh = self.client_ssh('192.168.188.91')
+        status, result1 = node0_ssh.exec_command(cmd_prefix1[1])
+        self.assertIn("MAC Address", result0, "peer_bmc can not Obtain lan")
+        self.assertIn("MAC Address", result1, "peer_bmc can not Obtain lan")
+        self.assertNotEqual(result0, result1, "peer_bmc can not Obtain lan")
+
+        """
+        Test peer_bmc sensor list
+        """
+        node0_ssh = self.client_ssh('192.168.188.91')
+        status, result0 = node0_ssh.exec_command(cmd_prefix0[2])
+        node0_ssh = self.client_ssh('192.168.188.91')
+        status, result1 = node0_ssh.exec_command(cmd_prefix1[2])
+        self.assertNotEqual(result0, result1, "peer_bmc can not Obtain sensor")
+
+        """
+        Test peer_bmc sel list
+        """
+        node0_ssh = self.client_ssh('192.168.188.91')
+        status, result0 = node0_ssh.exec_command(cmd_prefix0[3])
+        node0_ssh = self.client_ssh('192.168.188.91')
+        status, result1 = node0_ssh.exec_command(cmd_prefix1[3])
+        self.assertNotEqual(result0, result1, "peer_bmc can not Obtain sensor")
+
+        """
+        Test peer_bmc off, on and cycle
+        """
+        node1_ssh = self.client_ssh('192.168.188.92')
+        os.system(cmd_prefix1[4])
+        PS_QEMU = "ps ax | grep qemu"
+        qemu_result = run_command(PS_QEMU, True,
+                                 subprocess.PIPE, subprocess.PIPE)[1]
+        self.assertNotIn("test1", qemu_result, "peer_bmc can not power off")
+
+        os.system(cmd_prefix1[5])
+        node1_ssh = self.client_ssh('192.168.188.92')
+        status, result = node1_ssh.exec_command('ls')
+        assert result
+
+        os.system(cmd_prefix1[6])
+        node1_ssh = self.client_ssh('192.168.188.92')
+        status, result = node1_ssh.exec_command('ls')
+        assert result
+
+        os.system("sed -i '$d' {}".format(emu_file_for_del0))
```

Comment:
not necessary to delete, since stop_node() will call  terminate_workspace() and it clear all.